src/resources/invoice.rs

```diff
@@ -115,57 +116,72 @@ pub struct InvoiceLineItem {
     pub proration: bool,
     pub quantity: Option<u64>,
     pub subscription: Option<String>,
-    pub subscription_item: Option<String>,
     #[serde(default)]
-    // NOTE: Missing in response to InvoiceLineItem create
     #[serde(rename = "type")]
-    pub item_type: String, // (invoiceitem, subscription)
+    pub type_: String, // (invoiceitem, subscription)
 }
 
-impl Identifiable for InvoiceLineItem {
-    fn id(&self) -> &str {
+impl Object for InvoiceLineItem {
+    type Id = InvoiceLineItemId;
+    fn id(&self) -> &Self::Id {
         &self.id
     }
+    fn object(&self) -> &'static str {
+        "line_item"
+    }
 }
 
 /// The resource representing a Stripe invoice.
 ///
 /// For more details see https://stripe.com/docs/api#invoice_object.
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct Invoice {
-    pub id: Option<String>, // id field is not present when retrieving upcoming invoices
+    pub id: Option<InvoiceId>, // id field is not present when retrieving upcoming invoices
+
     pub amount_due: u64,
-    pub application_fee: Option<u64>,
+    pub amount_paid: u64,
+    pub amount_remaining: u64,
+    pub application_fee_amount: Option<u64>,
     pub attempt_count: u64,
     pub attempted: bool,
-    pub charge: Option<String>,
-    pub closed: Option<bool>,
+    pub auto_advance: bool,
+    pub charge: Option<Expandable<Charge>>,
+    pub created: Timestamp,
     pub currency: Currency,
-    pub customer: String,
-    pub date: Timestamp,
+    pub customer: Expandable<Customer>,
+    pub due_date: Timestamp,
     pub description: Option<String>,
     pub discount: Option<Discount>,
     pub ending_balance: Option<i64>,
-    pub forgiven: bool,
     pub lines: List<InvoiceLineItem>,
     pub livemode: bool,
     pub metadata: Metadata,
     pub next_payment_attempt: Option<Timestamp>,
+    pub number: String,
     pub paid: bool,
     pub period_end: Timestamp,
     pub period_start: Timestamp,
     pub receipt_number: Option<String>,
     pub starting_balance: i64,
     pub statment_descriptor: Option<String>,
     pub subscription: Option<String>,
-    pub subscription_proration_date: Option<Timestamp>,
     pub subtotal: i64,
     pub tax: Option<i64>,
     pub tax_percent: Option<f64>,
     pub total: i64,
     pub webhooks_delivered_at: Option<Timestamp>,
 }
 
+impl Object for Invoice {
+    type Id = Option<InvoiceId>; // an invoice id can be none if it is an _upcoming_ invoice
```

Comment:
bleh. Does this mean `Invoice` can't be paginated?

Reply:
It works the same as before, Invoice isn't `Paginated` but there is the same custom `List` impl below